### PR TITLE
fix(aws-lambda-event-sources): unsupported properties for SelfManagedKafkaEventSource and ManagedKafkaEventSource

### LINKED ISSUE #17934: (aws-lambda-event-sources): kafka event source unsupported properties
### General Issue

Kafka event source properties includes unsupported properties

### The Question

The SelfManagedKafkaEventSourceProps interface includes properties such as: bisectBatchOnError, retryAttempts, maxBatchingWindow, onFailure, maxRecordAge, and parallelizationFactor however, per API docs ([ref](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/lambda/create-event-source-mapping.html)), these properties are only supported by Kinesis and Dynamo Stream event sources.
This causes confusion during deployment as it appears these features are available, but the deploy fails.

### CDK CLI Version

1.134.0 (build dd5e12d)

### Framework Version

1.134.0

### Node.js Version

v14.18.2

### OS

Debian

### Language

Typescript

### Language Version

_No response_

### Other information

Attempting to deploy returns an error:
`9:25:56 PM | CREATE_FAILED        | AWS::Lambda::EventSourceMapping | test...B0194DF1
Resource handler returned message: "Invalid request provided: Unsupported 'maximumRetryAttempts' parameter for given event source mapping type. (Service: Lambda, Status Code: 400, Request ID: 2f6..., Ext
ended Request ID: null)" (RequestToken: 324..., HandlerErrorCode: InvalidRequest)`

### PR DESCRIPTION
**Pull request by zehsor**
This PR fixes a bug in the CDK where some `kafkaEventSource` properties are actually unsupported. These properties exist only for kinesis and dynamodb streams. The existing KafkaEventSourceProps Interface erroneously extends an interface that includes kinesis and dynamodb specific properties. This PR separates these properties into a `Base` interface with shared stream properties for all 3, as well as an interface for `kinesis` and `dynamodb` specific properties. 

Unit testing unavailable because the scope of the PR is to remove properties. It is enough to ensure that current tests still succeed.

We are allowing the breaking changes specified in  `allowed-breaking-changes.txt` because they never worked in the first place.

Fixes #17934.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by iwt-philipzeh** [comment]
@kaizen3031593 not trying to interrupt or stress you, but is there anything i can do to help you review this? :)

File: packages/@aws-cdk/aws-lambda-event-sources/lib/kafka.ts [hunk 1]
```diff
@@ -13,7 +13,8 @@ import { Construct } from '@aws-cdk/core';
 /**
  * Properties for a Kafka event source
  */
-export interface KafkaEventSourceProps extends StreamEventSourceProps {
+export interface KafkaEventSourceProps extends BaseStreamEventSourceProps{
+
```
Comment (kaizen3031593) [inline packages/@aws-cdk/aws-lambda-event-sources/lib/kafka.ts:17]: excess new line

File: packages/@aws-cdk/aws-lambda-event-sources/lib/stream.ts [hunk 2]
```diff
@@ -45,6 +70,7 @@ export interface StreamEventSourceProps {
    */
   readonly maxRecordAge?: Duration;
 
+
```
Comment (kaizen3031593) [inline packages/@aws-cdk/aws-lambda-event-sources/lib/stream.ts:73]: Why the extra new line here? I think there are 4 places in this section that have 2 new lines, ideally you help me get rid of them all.

File: packages/@aws-cdk/aws-lambda-event-sources/lib/stream.ts [hunk 3]
```diff
@@ -15,25 +15,50 @@ export interface StreamEventSourceProps {
...
-  readonly onFailure?: lambda.IEventSourceDlq;
+  readonly maxBatchingWindow?: Duration;
+
+  /**
+   * If the stream event source mapping should be enabled.
+   *
+   * @default true
+   */
+  readonly enabled?: boolean;
+}
```
Comment (kaizen3031593) [inline packages/@aws-cdk/aws-lambda-event-sources/lib/stream.ts:50]: There should be a new line between lines 50 and 51

**Review by kaizen3031593 — CHANGES_REQUESTED** [review]
This looks great @zehsor! I'm fine with the breaking change (yes, it is) because it never would have worked in the first place. And no need to test -- I'll mark the PR as exempt. 

I have a few nits involving new lines that I normally would go in and do myself, but you haven't configured the PR to allow me to do this (next time, try PRing from a branch of your fork and the default permissions should allow maintainer contributions). Since I can't go in and change them myself, I'm going to bother you and have you remove them. Then, I'll approve the PR. Thanks for the contribution!

**Comment by zehsor** [comment]
Thanks @kaizen3031593 for the great feedback. I think the formatting issue happened due to trying out different things in the beginning, sorry for that. As i usually work with gitlab i learned something new about github pull requests, will follow your instructions next time :)

**Review by kaizen3031593 — APPROVED** [review]

**Pull request merged**